File: pkg/rpc/generate/typesystem/testdata/simple_enum/expected.go
package models

// Color - Available colors
type Color string

const (
	// Red color
	ColorRed Color = "red"
	// Green color
	ColorGreen Color = "green"
	// Blue color
	ColorBlue Color = "blue"
)

// Valid returns true if the Color value is valid
func (e Color) Valid() bool {
	switch e {
	case ColorRed, ColorGreen, ColorBlue:
		return true
	default:
		return false
	}
}

File: pkg/rpc/generate/typesystem/testdata/simple_enum/expected.cs
namespace MyApp.Models
{
    /// <summary>
    /// Available colors
    /// </summary>
    public static class Color
    {
        /// <summary>Red color</summary>
        public const string Red = "red";
        /// <summary>Green color</summary>
        public const string Green = "green";
        /// <summary>Blue color</summary>
        public const string Blue = "blue";

        /// <summary>
        /// Returns true if the value is a valid Color
        /// </summary>
        public static bool IsValid(string value)
        {
            switch (value)
            {
                case "red":
                case "green":
                case "blue":
                    return true;
                default:
                    return false;
            }
        }
    }
}
